#VRML_SIM R2025a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/floors/protos/RectangleArena.proto"

WorldInfo {
}
Viewpoint {
  orientation -0.02161209007411386 0.9982770613698206 0.05455113477705042 0.7557311200818657
  position -1.0318273524712778 0.29889365647343613 0.5337240149940359
}
TexturedBackground {
  texture "mars"
}
TexturedBackgroundLight {
}
RectangleArena {
  translation 0 0 -0.19
}
Robot {
  children [
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 1.732 1 0
        anchor -0.0519617 -0.03 -0.05
      }
      device [
        RotationalMotor {
          name "motor_BL"
        }
      ]
      endPoint Solid {
        translation -0.0519617 -0.03 -0.05
        rotation -0.37796194289203355 -0.6546559010851013 0.6546529010855546 2.4188553071795864
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
            }
            geometry Cylinder {
              height 0.01
              radius 0.02
            }
          }
        ]
        name "solid(2)"
        boundingObject USE wheel
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis -1.732 1 0
        anchor 0.0519613 -0.030000100000000002 -0.05
      }
      device [
        RotationalMotor {
          name "motor_BR"
        }
      ]
      endPoint Solid {
        translation 0.0504646 -0.0301242 -0.0507514
        rotation -0.3760805060590644 0.6553688793079638 0.6550229652439535 3.8604679595602716
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
            }
            geometry Cylinder {
              height 0.01
              radius 0.02
            }
          }
        ]
        name "solid(1)"
        boundingObject USE wheel
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 -1 0
        anchor 0 0.06 -0.05
      }
      device [
        RotationalMotor {
          name "motor_F"
        }
      ]
      endPoint Solid {
        translation 0 0.06 -0.05
        rotation 0.9999999999982307 -1.8811599999966718e-06 -9.589789999983032e-09 1.5708053071795867
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
            }
            geometry Cylinder {
              height 0.01
              radius 0.02
            }
          }
        ]
        boundingObject USE wheel
        physics Physics {
        }
      }
    }
    DEF body Shape {
      appearance PBRAppearance {
      }
      geometry Cylinder {
        height 0.1
        radius 0.05
      }
    }
  ]
  boundingObject USE body
  physics Physics {
  }
  controller "my_controller2"
}
